src/structures/ThreadChannel.js

```diff
@@ -423,13 +423,19 @@ class ThreadChannel extends Channel {
    * @readonly
    */
   get joinable() {
+    const permissions = this.permissionsFor(this.client.user);
+    if (!permissions) return false;
+    // This flag allows joining even if timed out
+    if (permissions.has(Permissions.FLAGS.ADMINISTRATOR, false)) return true;
+
     return (
       !this.archived &&
       !this.joined &&
-      this.permissionsFor(this.client.user)?.has(
+      permissions.has(
         this.type === 'GUILD_PRIVATE_THREAD' ? Permissions.FLAGS.MANAGE_THREADS : Permissions.FLAGS.VIEW_CHANNEL,
         false,
-      )
+      ) &&
+      this.guild.me.communicationDisabledUntilTimestamp < Date.now()
```

Comment:
You can always join threads, regardless of whether you're timed out or not

Reply:
Thank you for pointing this out! I have verified this and removed this check.